packages/orama/src/id-shortener.ts

```diff
@@ -0,0 +1,54 @@
+import { IdStore } from './types.js'
+
+const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()'
+
+export function generateShortId(last: string) {
+  if (last === '') {
+    return characters[0]
+  }
+
+  const lastGeneratedIdLastIndex = last.length - 1
+
+  const lastChar = last[lastGeneratedIdLastIndex]
+
+  const charIndex = characters.indexOf(lastChar)
+
+  if (charIndex === characters.length - 1) {
+    return last + characters[0]
+  }
+
+  return last.substring(0, lastGeneratedIdLastIndex) + characters[charIndex + 1]
+}
+
+export function generateUniqueShortId(db: IdStore, original: string): string {
+  if (Object.hasOwn(db.originalToShort, original)) {
+    return db.originalToShort[original]
+  }
+
+  const newId = generateShortId(db.lastShort)
+
+  db.originalToShort[original] = newId
+
+  db.lastShort = newId
+
+  db.shortToOriginal[newId] = original
+
+  return newId
+}
+
+export function getOriginalId(db: IdStore, id: string) {
+  return db.shortToOriginal[id]
+}
+
+export function getGeneratedId(db: IdStore, id: string) {
+  return db.originalToShort[id]
+}
+
+export function createNewIdDatabase() {
+  return {
+    lastShort: '',
```

Comment:
This could be an array of integers: the integers are the indexes of `characters`. In this way, the implementation of `generateShortId` should be more straightforward.

Something like:
```ts
function buildString(indexes: number[]) {
  return indexes.map(i => characters[i]).join('')
}

function updatePreviousIndex(indexes: number[], i: number) {
  if (number === -1) // Not possible {
    return
  }

  if (indexes[i] === characters.length) {
    updatePreviousIndex(indexes, i - 1)
  }

  number[i] ++
}

export function generateShortId(indexes: number[]) {
  if (indexes[indexes.length-1] === characters.length) {
    updatePreviousIndex(indexes, indexes.length - 1)
    indexes.push(0)
  }

  return buildString(indexes)
}
```

I don't know if it is more readable. WDYT?

Reply:
Great idea!! This saves a lot of space 🙌

Reply:
Actually I'm not sure it would make any difference

Reply:
You are right. Too complex logic without any reason. OOOPS